src/desktop/analytics/helpers.ts

```diff
@@ -0,0 +1,63 @@
+import { omit } from "lodash"
+
+/**
+ * Format and fire events triggered via react-tracking
+ */
+export const trackEvent = (data: any, options: object = {}) => {
+  const actionName = data.action || data.action_type
+  const trackingData = omit(data, ["action_type", "action"])
+
+  if (actionName) {
+    // FIXME: is this still used?
+    if (actionName === "Clicked read more") {
+      // Send Reaction's read more as a page view
+      return onClickedReadMore(data)
+    }
+    // Grab referrer from our trackingMiddleware in Reaction, since we're in a
+    // single-page-app context and the value will need to be refreshed on route
+    // change. See: https://github.com/artsy/force/blob/master/src/v2/Artsy/Analytics/trackingMiddleware.ts
+    const referrer = window.analytics.__artsyClientSideRoutingReferrer
+    let trackingOptions = options
+
+    if (referrer) {
+      trackingOptions = {
+        page: {
+          referrer,
+        },
+      }
+    }
+    // Send event to segment
+    window.analytics.track(actionName, trackingData, trackingOptions)
+  } else {
+    console.error(
+      `Unknown analytics schema being used: ${JSON.stringify(data)}`
+    )
+  }
+}
+
+export const onClickedReadMore = data => {
+  const pathname = data.pathname || location.pathname
+  const href = window.sd.APP_URL + "/" + pathname
+  window.analytics.page(
+    { path: pathname },
+    { integrations: { Marketo: false } }
+  )
+  if (window.PARSELY) {
+    window.PARSELY.beacon.trackPageView({
+      url: href,
+      js: 1,
+      action_name: "infinite",
+    })
+  }
+  if (window.Sailthru) {
+    window.Sailthru.track({
+      domain: "horizon.artsy.net",
+      spider: true,
+      track_url: true,
+      url: href,
+      use_stored_tags: true,
+    })
+  }
+  // Return early because we don't want to make a Segment call for read more
+  return
```

Comment:
I understand that this was moved from another file. Would it be possible to remove these lines?

Reply:
We still need this one, double checked, it is so we fire only a `page` event rather than `track` when expanding an article.